Add some tests covering regex quantifiers and options

Tpt opened the pull request:
Based on XPath spec: https://www.w3.org/TR/xpath-functions/#regex-syntax

Issue #46

rubensworks commented:
@Tpt Is there a particular reason this was only added for sparql10 and not sparql11?

Tpt commented:
> @Tpt Is there a particular reason this was only added for sparql10 and not sparql11?

@rubensworks If I am not mistaken the `sparql11` folder only contains tests for features added by  SPARQL 1.1. For example BGP and basic operators (=, <, +...) tests are only in the `sparql10` directory. Because `REGEX` was introduced in SPARQL 1.0 I added the tests there for consistency.

rubensworks commented:
Oh, in that case I misunderstood how the tests are structured...

jitsedesmet commented on sparql/sparql10/regex/manifest.ttl:
```diff
@@ -56,3 +63,128 @@
+            qt:data   <regex-data-quantifiers.ttl> ] ;
+      mf:result  <regex-quantifier-counted-lower-upper-bounds.srx> .
+
+:regex-quantifier-counted-lower-bound a mf:QueryEvaluationTest ;
+      mf:name    "REGEX with an {,2} quantifier" ;
+      mf:action
+          [ qt:query  <regex-quantifier-counted-lower-bound.rq> ;
+            qt:data   <regex-data-quantifiers.ttl> ] ;
+      mf:result  <regex-quantifier-counted-lower-bound.srx> .
```
Hey,
@rubensworks tasked me with checking these tests in Comunica.
I noticed this entry is contained 2 times.

Is this intended?

Tpt replied:
Thank you so much for that and for spotting this error. This is indeed a mistake. Fixed.

jitsedesmet commented on sparql/sparql10/regex/manifest.ttl:
```diff
@@ -56,3 +63,128 @@
+            qt:data   <regex-data-quantifiers.ttl> ] ;
+      mf:result  <regex-dot-all.srx> .
+
+:regex-case-insensitive a mf:QueryEvaluationTest ;
```
Additionally, this entry is not listed on lines 13-19 :)

Tpt replied:
Indeed. My bad. Fixed.

jitsedesmet commented:
I noticed 2 more things when implementing the tests in Comunica.

First:
The result file `regex-no-metacharacters-case-insensitive.srx` is missing?

Second:

The following manifest entry sticks out
```
:regex-data-quantifier-optional a mf:QueryEvaluationTest ;
      mf:name    "REGEX with an ? quantifier" ;
      mf:action
          [ qt:query  <regex-quantifier-optional.rq> ;
            qt:data   <regex-data-quantifiers.ttl> ] ;
      mf:result  <regex-quantifier-optional.srx> .
```
The subject is :regex**-data-**quantifier-optional while it operates over the files `regex-quantifier-optional`. Is there a need for the mention of `-data-`?

Tpt commented:
>  The result file regex-no-metacharacters-case-insensitive.srx is missing?

Yes, out of laziness because it should return the same results as `regex-no-metacharacters.srx`

> The subject is :regex**-data-**quantifier-optional while it operates over the files regex-quantifier-optional. Is there a need for the mention of -data-?

None, it is an artifact or an older naming I removed late in the MR writing. Fixed.

>Anyway, thank you for addressing my previous remarks :)

With pleasure! Thank you for testing this MR.

Tpt commented on sparql/sparql10/regex/regex-case-insensitive.srx:
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0"?>
+<sparql>
```
Done!  Thank you for the so great suggestions that were so easy to apply!

Tpt commented on sparql/sparql10/regex/manifest.ttl:
```diff
@@ -10,6 +10,13 @@
+      :regex-no-metacharacters :regex-no-metacharacters-case-insensitive
+      :regex-start-end :regex-start-end-multiline
+      :regex-char-class-expression :regex-negative-char-class-expression
+      :regex-ignore-whitespaces :regex-ignore-whitespaces-class-expression
```
Done!

afs commented:
@tpt -- Do you want to squash the commits down to one? Or I can do in the GH interface? Or leave it as 8 commits for "rebase and merge" (this repo normally uses "rebase and merge").

Tpt commented:
@afs Done! Buf feel free to do it in GitHub interface the next times if you prefer.

afs commented:
> Oh, in that case I misunderstood how the tests are structured...



@rubensworks - I don't think we have a definite policy.

More tests is better. Structure can change.

My (mild) pref would be new test areas under sparql-current. In this case, there is already existing test area and splitting them across sparql-N seems less helpful.

It does mean we are not keeping the compliance tests for Sparql-VER exectly as used at the time as implementation reports but that could be quite a burden for not much value IMO. It's a tradeoff of more tests for sparql-current over a historical record of implementation reports.
